engines/wintermute/base/gfx/x/modelx.cpp

```diff
@@ -104,9 +104,13 @@ bool ModelX::loadFromFile(const Common::String &filename, ModelX *parentModel) {
 	uint32 fileSize = 0;
 	byte *buffer = BaseFileManager::getEngineInstance()->getEngineInstance()->readWholeFile(filename, &fileSize);
 
-	byte *dataFormatBlock = buffer + 8;
+	// the header of an .X file consists of 16 bytes
+	// bytes 9 to 12 contain a string which can be 'txt', 'bin', 'bzip, 'tzip', depending on the format
+	byte dataFormatBlock[4];
+	Common::copy(buffer + 8, buffer + 11, dataFormatBlock);
+	dataFormatBlock[3] = '\0';
```

Comment:
This will truncate "bzip" to "bzi\0", so the case on line 115 can never fire.

Reply:
Actually I just increased the buffer size now, didn't think about strncmp.